### PR TITLE
feat: Move VSI Quickstart DA code out of the examples folder

### PR DESCRIPTION
**Pull request by Soaib024**
### Description

Moved VSI Quickstart DA code from examples to `pattern/quickstart`

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 <!-- Remove the content in this H2 heading after completing the steps -->
 
-The landing zone module can be used to create a fully customizable VPC environment within a single region. The four following patterns are starting templates that can be used to get started quickly with Landing Zone. These patterns are located in the [patterns](/patterns/) directory.
+The landing zone module can be used to create a fully customizable VPC environment within a single region. The five following patterns are starting templates that can be used to get started quickly with Landing Zone. These patterns are located in the [patterns](/patterns/) directory.
 
-Each of these patterns creates the following infrastructure:
+Each of these patterns (except QuickStart) creates the following infrastructure:
```
Comment (ocofaigh) [inline README.md:15]: There may be other Quickstart patterns in the future, so I would rename this:
`except QuickStart` -> `except VSI QuickStart `

File: README.md [hunk 2]
```diff
@@ -25,16 +25,17 @@ Each of these patterns creates the following infrastructure:
 
 Each pattern creates the following infrastructure on the VPC:
 
+- The QuickStart VSI pattern deploys edge VPC with one VSI and a jump server VSI in the management VPC
```
Comment (ocofaigh) [inline README.md:28]: I would move this to the 2nd bullet point (so the standard VSI comes after quickstart vsi)

File: README.md [hunk 3]
```diff
@@ -25,16 +25,17 @@ Each of these patterns creates the following infrastructure:
...
 
-|  VPC pattern                   |  Virtual server pattern        |  Red Hat OpenShift pattern       | Mixed pattern                      |
-| ------------------------------ | ------------------------------ | -------------------------------- | ---------------------------------- |
-| [![VPC](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/reference-architectures/vpc.drawio.svg)](patterns/vpc/README.md) | [![VSI](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/reference-architectures/vsi-vsi.drawio.svg)](patterns/vsi/README.md) | [![ROKS](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/reference-architectures/roks.drawio.svg)](patterns/roks/README.md) |  [![Mixed](https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/.docs/images/mixed.png)](patterns/mixed/README.md) |
+| QuickStart pattern             |  VPC pattern                   |  Virtual server pattern          |  Red Hat OpenShift pattern       | Mixed pattern                      |
```
Comment (ocofaigh) [inline README.md:36]: `QuickStart pattern` -> `QuickStart VSI pattern`

File: examples/quickstart/catalogValidationValues.json.template [hunk 4]
```diff
@@ -1 +0,0 @@
-{"ibmcloud_api_key": $VALIDATION_APIKEY, "region": "eu-de", "resource_tags": $TAGS, "prefix": $PREFIX}
```
Comment (ocofaigh) [inline examples/quickstart/catalogValidationValues.json.template:1]: This file needs to be moved to `patterns/quickstart` too (looks like its missing)

**Review by ocofaigh — CHANGES_REQUESTED** [review]
I left a few comments. Also...
1. Can we rename the directory to `patterns/vsi-quickstart` please?
2. The directory needs to be updated in https://github.ibm.com/GoldenEye/operations/blob/master/catalog-onboard/scripts/product-mapping.yaml before this PR is merged, so please prepare a PR for that.

Thank you

File: tests/scripts/pre-validation-vsi-quickstart.sh [hunk 5]
```diff
@@ -2,4 +2,4 @@
 
 set -e
 
-./pre-validation-generate-ssh-key.sh ssh_key
+./pre-validation-generate-ssh-key.sh ssh_public_key
```
Comment (ocofaigh) [inline tests/scripts/pre-validation-vsi-quickstart.sh:5]: why was this changed? The variable name for quickstart is `ssh_key` so please change it back

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/508/files#r1274805408

File: tests/scripts/pre-validation-vsi-quickstart.sh (hunk 5 above)
Comment (Soaib024) [reply tests/scripts/pre-validation-vsi-quickstart.sh:5]: I have changed the variable name to `ssh_public_key` to make it inline with other patterns, as we are adding new pattern I thought this shouldn’t impact existing user, should I revert it back to `ssh_key`?

Comment (ocofaigh) [reply tests/scripts/pre-validation-vsi-quickstart.sh:5]: yes please revert it back - the VSI quickstart DA is already in production, so lets leave it as is

**Review by ocofaigh — CHANGES_REQUESTED** [review]
The `ibm_catalog.json` needs to be updated too (line 52)

**Review by ocofaigh — APPROVED** [review]
